### PR TITLE
feat: facebook custom audience app secret support

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

Adding app secret support

## What is the related Linear task?

Resolves INT-2075

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [x] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3357/test-report.html

**Review by anantjain45823 — APPROVED** [review]

**Review by sandeepdsvs — DISMISSED** [review]

File: src/v0/destinations/fb_custom_audience/transform.js [hunk 1]
```diff
@@ -105,6 +106,11 @@ const prepareResponse = (
 
   prepareParams.access_token = accessToken;
 
+  if (isDefinedAndNotNullAndNotEmpty(appSecret)) {
+    prepareParams.appsecret_time = Math.floor(Date.now() / 1000); // Get current Unix time in seconds
```
Comment (utsabc) [inline src/v0/destinations/fb_custom_audience/transform.js:110]: get the `Date.now()` and store it in a variable and use same in `appsecret_time` and in the `generateAppSecretProof`.

Current logic might work well with seconds precision but in future if we add milliseconds precision the timestamps might vary

File: test/integrations/destinations/fb_custom_audience/processor/data.ts [hunk 2]
```diff
@@ -54419,4 +54424,148 @@ export const data = [
       },
     },
   },
-];
+  {
+    name: 'fb_custom_audience',
+    description: 'Type NA ',
+    feature: 'processor',
+    module: 'destination',
+    version: 'v0',
```
Comment (utsabc) [inline test/integrations/destinations/fb_custom_audience/processor/data.ts:54432]: Lets add description and other details for the newly added test cases for ease of future migration of test cases

Comment (shrouti1507) [reply test/integrations/destinations/fb_custom_audience/processor/data.ts]: added

File: src/v0/destinations/fb_custom_audience/transform.js (hunk 1 above)
Comment (shrouti1507) [reply src/v0/destinations/fb_custom_audience/transform.js]: addressed the comment

**Review by utsabc — APPROVED** [review]

**Review by anantjain45823 — APPROVED** [review]

**Pull request merged**